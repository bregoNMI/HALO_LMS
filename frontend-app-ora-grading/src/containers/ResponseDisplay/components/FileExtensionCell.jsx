import React from 'react';
import PropTypes from 'prop-types';

export const FileExtensionCell = ({ value }) => (
  <div className="text-truncate">{value.split('.')?.pop().toUpperCase()}</div>
);

FileExtensionCell.defaultProps = {};

FileExtensionCell.propTypes = {
  value: PropTypes.string.isRequired,
};

export default FileExtensionCell;
