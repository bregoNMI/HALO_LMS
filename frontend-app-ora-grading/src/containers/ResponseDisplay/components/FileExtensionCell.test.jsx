import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import FileExtensionCell from './FileExtensionCell';

describe('FileExtensionCell', () => {
  describe('component', () => {
    const props = {
      value: 'file_name.with_extension.pdf',
    };
    let el;
    beforeEach(() => {
      el = shallow(<FileExtensionCell {...props} />);
    });
    test('snapshot', () => {
      expect(el.snapshot).toMatchSnapshot();
    });

    describe('behavior', () => {
      test('content', () => {
        expect(el.instance.children[0].el).toEqual('PDF');
      });
    });
  });
});
