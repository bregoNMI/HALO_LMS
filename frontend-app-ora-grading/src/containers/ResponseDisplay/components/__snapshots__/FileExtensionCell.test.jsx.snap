// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileExtensionCell component snapshot 1`] = `
<div
  className="text-truncate"
>
  PDF
</div>
`;
